### PR TITLE
Update repo documentation for Gromacs

### PR DESCRIPTION
**Pull request by HubLot**
This PR update the README with the Gromacs information.
I added also a documentation (`devel-tools/create_gromacs-colvars-releases.md`) on how to create Gromacs-colvars release in the repository https://github.com/Colvars/gromacs/

### CODE REVIEW

File: devel-tools/create_gromacs-colvars-releases.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Create patched colvars gromacs versions
+
+Steps to create a release archive of a Gromacs version with colvars support.
```
Comment (jhenin) [inline devel-tools/create_gromacs-colvars-releases.md:3]: "Colvars" (capital 'C')

File: devel-tools/create_gromacs-colvars-releases.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+This is done on the Gromacs fork of Colvars organization: https://github.com/Colvars/gromacs
+
+All commands are done within the GROMACS source tree folder.
+
+If a colvars release branch already exists with a Colvars commit, go to step 3.
```
Comment (jhenin) [inline devel-tools/create_gromacs-colvars-releases.md:8]: Colvars

File: devel-tools/create_gromacs-colvars-releases.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+## 1. Create a derived branch from a release one
+
+```
+git checkout -b release-XXXX-colvars release-XXX
+```
+
+## 2. Apply, check and commit colvars patch
+
+This step implies that the colvars sources support the desired Gromacs version.
```
Comment (jhenin) [inline devel-tools/create_gromacs-colvars-releases.md:18]: Same... :-)

File: devel-tools/create_gromacs-colvars-releases.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+```
+# Applying patch
+/path/to/colvars/update-colvars-code.sh .
+```
+
+Next, is to make sure the patch works but compiling gromacs and eventually, run the colvars gromacs tests.
```
Comment (jhenin) [inline devel-tools/create_gromacs-colvars-releases.md:25]: and here as well

**Review by jhenin — APPROVED** [review]
Looks very good to me! With a few tiny edits as indicated.

**Comment by HubLot** [comment]
Thanks. I fixed those typos plus a few others.

**Pull request merged**